app/models/user.rb

```diff
@@ -356,10 +356,7 @@ def change_requests_owned
   #########################################################
 
   # Delegate methods to summary object
-  UserSummary.attribute_names.each do |name|
-    next if name == 'id' || name.end_with?('_id', '_at')
-    def_delegator :user_summary, name.to_sym, name.to_sym
-  end
+  delegate_missing_to :user_summary
```

Comment:
The call to `UserSummary.attribute_names` forced Rails to connect to the database to figure out the table structure. However, since this was happening in the class body, it got run as soon as the class was loaded, which takes place when `devise` is configured in the routes file.

This was relatively early and it didn't give the application a chance to finish initialization or run `after_initialize` hooks. It made the database connection a mandatory dependency to load the application, causing all Rake tasks to be unusable unless the connection can be established.

My suggested solution is to use `delegate_missing_to`, which avoids the dependency during startup and is more flexible with attribute-like methods (e.g. predicates such as `user_rep_raw?`).